data/net/minecraft/world/item/enchantment/Enchantment.mapping

```diff
@@ -1,55 +1,103 @@
+		COMMENT Calculates the amount of additional damage to deal when attacking with an item with the enchantment.
 		ARG 1 level
-		ARG 2 creatureType
+			COMMENT The level of the enchantment being used.
+		ARG 2 mobType
+			COMMENT The type of mob being damaged.
 	METHOD getDamageProtection (ILnet/minecraft/world/damagesource/DamageSource;)I
-		COMMENT Calculates the damage protection of the enchantment based on level and damage source passed.
+		COMMENT Calculates the amount of additional damage protection to provide when a user is damaged.
 		ARG 1 level
+			COMMENT The level of the enchantment being used.
 		ARG 2 source
+			COMMENT The source of the damage.
 	METHOD getDescriptionId ()Ljava/lang/String;
-		COMMENT Return the name of key in translation table of this enchantment.
+		COMMENT Gets the description Id of the enchantment. This is commonly used to create localization keys.
 	METHOD getFullname (I)Lnet/minecraft/network/chat/Component;
+		COMMENT Gets the name of the enchantment with the level appended to the end. The game will attempt to convert the letter to roman numerals.
+		COMMENT @return The name of the enchantment with the level appened to the end.
 		ARG 1 level
+			COMMENT The level of the enchantment.
 	METHOD getMaxCost (I)I
-		ARG 1 enchantmentLevel
+		ARG 1 level
 	METHOD getMaxLevel ()I
-		COMMENT Returns the maximum level that the enchantment can have.
+		COMMENT Gets the maximum level of the enchantment under normal circumstances.
 	METHOD getMinCost (I)I
 		COMMENT Returns the minimal value of enchantability needed on the enchantment level passed.
-		ARG 1 enchantmentLevel
+		ARG 1 level
 	METHOD getMinLevel ()I
-		COMMENT Returns the minimum level that the enchantment can have.
+		COMMENT Gets the minimum level of this enchantment under normal circumstances.
+	METHOD getOrCreateDescriptionId ()Ljava/lang/String;
+		COMMENT Gets the raw description Id of the enchantment or creates a new one using the registry name of the enchantment.
+		COMMENT @return The descriptionId or a newly created Id.
 	METHOD getRarity ()Lnet/minecraft/world/item/enchantment/Enchantment$Rarity;
-		COMMENT Retrieves the weight value of an Enchantment. This weight value is used within vanilla to determine how rare an enchantment is.
+		COMMENT Gets the rarity of the enchantment. This determines how rare the enchantment will be in enchantment pools such as the enchanting table and loot tables.
 	METHOD getSlotItems (Lnet/minecraft/world/entity/LivingEntity;)Ljava/util/Map;
-		COMMENT Gets map of all the entity's currently equipped gear that this enchantment can go on
-		ARG 1 livingEntity
+		COMMENT Creates a new map containing all items equipped by an entity in slots that the enchantment cares about. {@link slots} These items are not tested for having the enchantment.
```

Comment:
This link goes nowhere (`#` prefix missing). It also doesn't really fit with the rest of the comment. I would suggest
`Creates a new map containing all items equipped by an entity in {@linkplain #slots slots that the enchantment cares about}. These items are not tested for having the enchantment.`